src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -40,14 +40,14 @@ public function __construct(array $headers = array())
      */
     public function __toString()
     {
-        if (!$this->headers) {
+        if (!$headers = $this->all()) {
```

Comment:
`ResponsHeaderBag` overrides `all()` by including the additional cookie headers. `HeaderBag::all` still returns `$this->headers`... so it's not removed.

However internally it now uses `all()` (the API method) instead of `$headers` (the property) so those cookie headers are taken into account when using for example `HeaderBag::has`.